src/main/java/gyro/azure/keyvault/VaultAccessPolicy.java

```diff
@@ -0,0 +1,264 @@
+/*
+ * Copyright 2020, Perfect Sense, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package gyro.azure.keyvault;
+
+import java.util.HashSet;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+import com.microsoft.azure.management.keyvault.AccessPolicy;
+import com.microsoft.azure.management.keyvault.CertificatePermissions;
+import com.microsoft.azure.management.keyvault.KeyPermissions;
+import com.microsoft.azure.management.keyvault.Permissions;
+import com.microsoft.azure.management.keyvault.SecretPermissions;
+import com.microsoft.azure.management.keyvault.StoragePermissions;
+import com.microsoft.azure.management.keyvault.Vault;
+import com.microsoft.rest.ExpandableStringEnum;
+import gyro.azure.Copyable;
+import gyro.core.resource.Diffable;
+import gyro.core.resource.Output;
+import gyro.core.resource.Updatable;
+import gyro.core.validation.Required;
+
+public class VaultAccessPolicy extends Diffable implements Copyable<AccessPolicy> {
+
+    private String name;
+    private Set<String> certificatePermissions;
+    private Set<String> keyPermissions;
+    private Set<String> secretPermissions;
+    private Set<String> storagePermissions;
+    private String objectId;
+
+    /**
+     * A set of allowed certificate access permissions.
```

Comment:
Is there a list of allowed permissions? A link to some docs we can provide? 

I did some quick searching of the Key Vault docs and couldn't find what values should go here. The portal seems to imply it's a limited set such as "create", "import", etc.

Reply:
Updated the docs to have links to the full set of available options